src/services/jsonSchemaService.ts

```diff
@@ -562,18 +569,16 @@ export class JSONSchemaService implements IJSONSchemaService {
 					delete next.$ref;
 					if (segments[0].length > 0) {
 						// This is a reference to an external schema
-						openPromises.push(resolveExternalLink(next, segments[0], segments[1], parentSchemaURL, parentSchemaDependencies));
+						openPromises.push(resolveExternalLink(next, segments[0], segments[1], parentHandle));
 						return;
 					} else {
 						// This is a reference inside the current schema
 						if (!seenRefs.has(ref)) {
-							if (isSubSchemaRef(segments[1])) { // A $ref to a sub-schema with an $id (i.e #hello)
-								// Get the full URI for the current schema to avoid matching schema1#hello and schema2#hello to the same
-								// reference by accident
-								const fullId = reconstructRefURI(parentSchemaURL, segments[1]);
-								tryMergeSubSchema(fullId, next);
+							const id = segments[1];
+							if (id !== undefined && isSubSchemaRef(id)) { // A $ref to a sub-schema with an $id (i.e #hello)
+								tryMergeSubSchema(next, id, handle);
```

Comment:
Pay attention you've replaced `parentSchemaURL` with `handle`. 
Therefore local refs in an external schemas wont work.

Reply:
Thanks, created #121